12/umbraco-cms/reference/api-versioning-and-openapi.md

```diff
@@ -0,0 +1,346 @@
+---
+description: How to use API versioning and OpenAPI (Swagger) for your own APIs.
+---
+
+# API versioning and OpenAPI (Swagger)
+
+Umbraco 12 ships with Swagger to document the Content Delivery API. Swagger and the Swagger UI is available at `{yourdomain}/umbraco/swagger`. For security reasons, both are disabled in non-production environments.
+
+Due to the way OpenAPI works within ASP.NET Core, we have to apply some configurations in a global scope. If your Umbraco site used Swagger previous to Umbraco 12, these global configurations may interfere with your setup.
+
+In this article we'll explore concrete solutions to overcome challenges with the global configurations.
+
+{% hint style="info" %}
+Umbraco uses [Swashbuckle.AspNetCore](https://github.com/domaindrivendev/Swashbuckle.AspNetCore/) to handle Swagger and the Swagger UI.
+
+If you have been using [NSwag](https://github.com/RicoSuter/NSwag) previous to Umbraco 12, chances are your Swagger setup will continue to work in Umbraco 12+ without any changes. Swashbuckle.AspNetCore and NSwag can coexist within the same site, as long as there are no conflicting routes between the two.
+
+That being said, it would be sensible to consider migrating your API documentation to Swashbuckle.AspNetCore. This way you can avoid having multiple dependencies that perform the same tasks.
+{% endhint %}
+
+## API versioning
+
+The Umbraco APIs rely on having the requested API version as part of the URL. If you prefer a different versioning for your own APIs, you can setup alternatives while still preserving the functionality of the Umbraco API.
+
+The following code sample illustrates how you can use a custom header to pass the requested API version to your own APIs.
+
+{% code title="Changing the API versioning" %}
```

Comment:
When we use the Code Title, we use it for the file name.
Would that make sense to do in this case as well?

Reply:
I'll update all the code titles where it makes sense, or remove them where it does not.

Reply:
Done 👍